core/src/replay_stage.rs

```diff
@@ -2640,14 +2640,44 @@ impl ReplayStage {
                     .expect("Bank fork progress entry missing for completed bank");
 
                 let replay_stats = bank_progress.replay_stats.clone();
-                let r_replay_stats = replay_stats.read().unwrap();
+                let mut replay_stats = replay_stats.write().unwrap();
```

Comment:
Is this only mutable for the accumulate on line 2651? Maybe we can keep the write lock nested down below and then grab a read lock after

Reply:
yep. thanks for nice suggestion! well, i overlooked proper locking granularity thinking...: 3a7810c18b8dddc3fdd8849d600de2136871182e